### ISSUE TITLE
How to only check packages listed in versionGroups -- or how to have versionGroups match packages beyond the syncpackrc filter property?

### ISSUE THREAD

**Issue by dsilvasc** [JamieMason/syncpack#97]
## Description

Right now if you specify version groups with specific packages and/or with package patterns (with wildcards), syncpack still checks all packages unless you list each package again in the `filter` property of syncpackrc.  Eventually, if you want to only match packages in version groups, the filter regex gets fairly unwieldy.

If there were a way to only check packages included in version groups, it would be easier to gradually onboard to syncpack.

## Suggested Solution

Would it be possible to have a mode that only matches through version groups, not the filter property?  Maybe by setting the filter to the empty string or false?

Another way to look at it would be that versionGroup packages wouldn't be filtered.

**Comment by dsilvasc** [comment]
Likewise for the top level `dev`, `peer`, `prod` properties in syncpackrc.

**Comment by JamieMason** [comment]
Thanks @dsilvasc, could you make a small repro to demonstrate this please? I'll need to see the scenario you're facing so I can dig into it.

Dependencies are assigned to versionGroups in https://github.com/JamieMason/syncpack/blob/c6db8786f9a117eb9d1d334edc7dfa70b036eeaf/src/lib/get-input/get-instances.ts#L34

_In theory_ it shouldn't be possible for a given dependency to belong to more than one versionGroup because they're looped over once and each dependency is partitioned into the first group it matches. If version groups aren't being used or a dependency matches none of them, it goes into a final catch-all group. 

Maybe something else is going on, we will see, if you can create a reproduction please with expected vs actual behaviour, I'll hopefully be able to figure it out using that.

**Comment by dsilvasc** [comment]
Here's a repro: https://github.com/dsilvasc/repro-syncpack-97

Expected:

```
$ npx syncpack list-mismatches
= Version Group 2 ===============================================================
- webpack: ^5.74.0 is the highest valid semver version in use
  5 in devDependencies of packages/a/package.json
  ^5.74.0 in devDependencies of packages/b/package.json
  5 in devDependencies of packages/c/package.json
```

Actual:

```
$ npx syncpack list-mismatches
- react:  is the highest valid semver version in use
  >16 in peerDependencies of packages/a/package.json
  >17 in peerDependencies of packages/c/package.json
- react-dom:  is the highest valid semver version in use
  >16 in peerDependencies of packages/a/package.json
  >17 in peerDependencies of packages/c/package.json
```

**Comment by JamieMason** [comment]
Hey @dsilvasc, I just took a look at the repro. As I think you might have mentioned: `filter` applies globally, when it is set then only packages whose names match that RegEx will be included.

I just want to check before I do it, would something like this work for you?

```diff
versionGroups:
  - packages: ['**']
    dependencies: ['webpack']
  - packages: ['**']
    dependencies: ['react', 'react-dom']
    dependencyTypes: ['dependencies', 'devDependencies']
+  - packages: ['**']
+    dependencies: ['**']
+    isIgnored: true
```

What this would do is create a catch-all version group which ignores every package which does not match any of the preceding groups.

Thanks.

**Comment by dsilvasc** [comment]
Hi @JamieMason, thank you for looking at this -- I think that new flag does capture it.

**Comment by JamieMason** [comment]
This was released in [8.4.10](https://github.com/JamieMason/syncpack/releases/tag/8.4.10), any problems let me know 👍🏻

**Comment by dsilvasc** [comment]
Thank you!
